File: 0409-longest-palindrome/0409-longest-palindrome.py
class Solution:
    def longestPalindrome(self, s: str) -> int:
        freq = {}
        count = 0
        singleele = False
        for ch in s:
            freq[ch] = freq.get(ch,0)+1
        print(freq)
        for key in freq:
            if(freq[key]%2 == 0):
                count+=freq[key]
            else:
                if count%2==0:
                    count+=freq[key]
                else:
                    count+=freq[key]-1

        return count        

File: 0409-longest-palindrome/0409-longest-palindrome.java
class Solution {
    public int longestPalindrome(String s) {
        int freq[] = new int[58];
        for(char c:s.toCharArray()){
            freq[c-'A']++;
        }
        int count=0;
        for(int i=0;i<freq.length;i++){
            int val = freq[i];
           if(val%2==0){
            count+=val;
           }else{
            if(count%2==0){
                count+=val;
            }else{
                count+=val-1;
            }
           }
        }
        return count;
    }
}